### PR TITLE
[JENKINS-37604] API to get a link to the branch on Github

### PR DESCRIPTION
**Pull request by i386**
# Description

API to get a link to the branch on Github. Not possible to write ATH as we do not have any live tests that work against Github and have provided a unit test instead.

See [JENKINS-37604](https://issues.jenkins-ci.org/browse/JENKINS-37604).

# Submitter checklist
- [x] Link to JIRA ticket in description, if appropriate.
- [x] Change is code complete and matches issue description
- [x] Appropriate unit or acceptance tests or explanation to why this change has no tests
- [ ] Reviewer's manual test instructions provided in PR description. See Reviewer's first task below.
- [ ] Ran Acceptance Test Harness against PR changes.

# Reviewer checklist
- [ ] Run the changes and verified the change matches the issue description
- [ ] Reviewed the code
- [ ] Verified that the appropriate tests have been written or valid explanation given

@reviewbybees

### CODE REVIEW

File: blueocean-pipeline-api-impl/src/main/java/io/jenkins/blueocean/rest/impl/pipeline/BranchImpl.java [hunk 1]
```diff
@@ -39,16 +42,31 @@ public BranchImpl(Job job, Link parent) {
...
+                om != null ? om.getObjectUrl() : null,
+                om != null ? om.getObjectDisplayName() : null,
+                cm != null ? cm.getContributor() : null
+            );
         }
         return null;
     }
 
+    @Exported(name = URL)
+    public String getUrl() {
```
Comment (i386) [inline blueocean-pipeline-api-impl/src/main/java/io/jenkins/blueocean/rest/impl/pipeline/BranchImpl.java]: Added in this PR

File: blueocean-pipeline-api-impl/src/test/java/io/jenkins/blueocean/rest/impl/pipeline/MultiBranchTest.java [hunk 2]
```diff
@@ -79,6 +84,16 @@ private boolean runAllTests() {
 
 
     @Test
+    public void testGetURL() {
```
Comment (i386) [inline blueocean-pipeline-api-impl/src/test/java/io/jenkins/blueocean/rest/impl/pipeline/MultiBranchTest.java]: Added in this PR

**Comment by i386** [comment]
@vivek this is now ready for review.

**Comment by vivek** [comment]
@i386 LGTM, however we should hold on to considering https://github.com/jenkinsci/blueocean-plugin/pull/724?

**Comment by i386** [comment]
@vivek I believe that would be wise yes

**Comment by vivek** [comment]
Closing this as it has been implemented in https://github.com/jenkinsci/blueocean-plugin/pull/707.

**Pull request closed without merging**